addons/product/data/product_demo.xml

```diff
@@ -221,6 +221,17 @@
             },]"/>
         </function>
 
+        <!--
+        This will delete the variant product_product_4d that is created from the template with the 2x2 variants
+        -->
```

Comment:
`product_product_4d` doesn't mean anything anymore since the xml_id was removed.
I think you can remove this comment as it'll be simply confusing for whoever reads this comment without knowing the past state of the code.